### PR TITLE
Environment sanity checks in script/server #3824

### PR DESCRIPTION
**Pull request by aka001**

### CODE REVIEW

File: script/server [hunk 1]
```diff
@@ -61,6 +61,22 @@ then
   fatal "config/diaspora.yml is missing! Copy over config/diaspora.yml.example to config/diaspora.yml and edit it properly!"
 fi
 
+# Check if detached head state
+git status | grep 'Changes not staged for commit' &> /dev/null
+if [ $? ]
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
```
Comment (jhass) [inline script/server]: You can't detect detached head status by seeing if there are unstaged changes. Detached head status happens when you do for example `git checkout v0.3.0.0`, that is checkout anything other than a branch.

Also the main point was to ensure creating the tarball below doesn't fail in that state.

File: script/server [hunk 2]
```diff
@@ -61,6 +61,22 @@ then
...
+
+# Check if bundle is complete
+bundle check | grep 'Install missing gems' > /dev/null
+if [ $? ]
+then
+  warning "Your bundle is not up to date, run the command \"bundle install\""
+else
+  echo "Your bundle is up to date"
```
Comment (jhass) [inline script/server]: No need for this message.

File: script/server [hunk 3]
```diff
@@ -61,6 +61,22 @@ then
...
+if [ $? ]
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
+
+# Check if bundle is complete
+bundle check | grep 'Install missing gems' > /dev/null
+if [ $? ]
+then
+  warning "Your bundle is not up to date, run the command \"bundle install\""
```
Comment (jhass) [inline script/server]: This should be more than a warning, it should fail.

File: script/server [hunk 4]
```diff
@@ -61,6 +61,22 @@ then
...
 
+# Check if detached head state
+git status | grep 'Changes not staged for commit' &> /dev/null
+if [ $? ]
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
+
+# Check if bundle is complete
+bundle check | grep 'Install missing gems' > /dev/null
```
Comment (jhass) [inline script/server]: > Description:
>  Check searches the local machine for each of the gems requested in the Gemfile. If all gems are found, Bundler prints a success message and exits with a status of 0. If 
  not, the first missing gem is listed and Bundler exits status 1.

There's no need for the grep, just checking the return code is enough.

**Comment by jhass** [comment]
The current script works even when git is not available or if the current directory is not a git repository, please retain those properties.

**Comment by aka001** [comment]
@jhass Without 'git' how we can determine whether the repository is in detached head state?
I have fixed the bundle check issue.

**Comment by jhass** [comment]
That point is really mainly for not letting the tar generation fail in that case, that is add it to the conditions that skip it.

File: script/server [hunk 5]
```diff
@@ -61,6 +61,18 @@ then
...
 fi
 
+# Check if detached head state
+if git status | grep 'Changes not staged for commit' > /dev/null ;
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
+
+# Check if bundle is complete
+if !( bundle check ) ;
```
Comment (jhass) [inline script/server]: No need for the parens or the semicolon. you also should suppress its output IMO.

**Comment by aka001** [comment]
I have updated the pr. Kindly review it.

File: script/server (hunk 1 above)
Comment (jhass) [inline script/server]: The above comment still holds.

File: script/server [hunk 6]
```diff
@@ -61,6 +61,33 @@ then
...
+fi
+
+# Check if bundle is complete
+if ! bundle check > /dev/null
+then
+  fatal "Your bundle is not up to date, run the command \"bundle install\""
+fi
+
+# Check if database is migrated and correct gemset
+if rake db:migrate:status | grep 'down' > /dev/null ;
```
Comment (jhass) [inline script/server]: In the original issue I say

> but if we can't find a way that takes less than 3 or 5 seconds its probably not worth it.

On my server:
```
$ time bin/rake db:migrate:status
...
real    0m7.318s
user    0m5.760s
sys     0m1.200s
```

File: script/server [hunk 7]
```diff
@@ -61,6 +61,33 @@ then
...
+then
+  warning "Your database is not migrated, run the command \"rake db:migrate\""
+elif rake db:migrate:status | grep 'rake aborted' > /dev/null ;
+then
+  warning "The gemset is not correct"
```
Comment (jhass) [inline script/server]: This really meant to check if RVM is available that the currently activated gemset is named diaspora, not whether "sokmething went wrong while running a rake command".

File: script/server [hunk 8]
```diff
@@ -61,6 +61,33 @@ then
...
+# Check if detached head state
+if git status | grep 'Changes not staged for commit' > /dev/null ;
+then
+  warning "Your changes are not staged for commit. Kindly stash or commit them"
+fi
+
+# Check if RVM available
+if rvm -v | grep 'No command '"'"'rvm'"'"' found' > /dev/null ;
+then
+  warning "RVM is not installed on the system. Please install it"
```
Comment (jhass) [inline script/server]: We recommend, not require RVM.

**Comment by jhass** [comment]
Superseded by #3824 

Thank you!

**Comment by Flaburgan** [comment]
#3824 is the issue, this PR is superseded by https://github.com/diaspora/diaspora/pull/5891 ;)

**Comment by jhass** [comment]
Ups, thanks!

**Pull request closed without merging**